src/libtiled/hexagonalrenderer.cpp

```diff
@@ -289,6 +318,9 @@ void HexagonalRenderer::drawTileLayer(QPainter *painter,
 
     CellRenderer renderer(painter, CellRenderer::HexagonalCells);
 
+    const int endX = map()->infinite() ? layer->bounds().width() : layer->width();
```

Comment:
I think the `endX` should be `layer->bounds().right() - layer->x() + 1` in case of infinite maps. Similar for Y.